packages/eas-cli/src/update/utils.ts

```diff
@@ -74,3 +78,60 @@ export function ensureValidVersions(exp: ExpoConfig, platform: RequestedPlatform
     throw error;
   }
 }
+
+export async function checkDeprecatedChannelConfigurationAsync(
```

Comment:
you are mixing here to responsibilities, this function should either:
- check if current config is deprecated and return true/false without logging anything. I would also expect in that case that function will be named `isUsingDeprcatedChannelCOnfiguration` or sth like that
- handle the situation(e.g. print warning and return) and do not return any values